app/controllers/api/artists_controller.rb

```diff
@@ -7,7 +7,7 @@ def index
 
     def show
         @artist = Artist
-            .includes(:albums, collab_songs: [{album: :album_artist}, :collab_artists])
+            .includes(albums:[:tracks], collab_songs: [:collab_artists])
```

Comment:
I have in development a (Ruby gem called bullet)[https://github.com/flyerhzm/bullet] that helps me avoid n+1 queries by telling me when I should ("eager-load" associated data)[https://blog.protein.tech/understanding-and-solving-n-1-queries-in-ruby-on-rails-98a0b25160b] from other models during an ActiveRecord query. Here, it told me I was eager-loading unneeded information.